pkg/backend/filestate/store_test.go

```diff
@@ -395,3 +396,90 @@ func TestProjectReferenceStore_List(t *testing.T) {
 		})
 	}
 }
+
+func TestProjectReferenceStore_ProjectExists(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		desc string
+
+		// List of file paths relative to the storage root
+		// that should exist before ListReferences is called.
+		files []string
+
+		// Project name that should exist before ProjectExists is called.
+		projectName string
+
+		// Result that should be returned by ProjectExists.
+		exist bool
+
+		// Err that should be returned by ProjectExists.
+		err error
+	}{
+		{
+			desc:        "empty string project name",
+			files:       []string{},
+			projectName: "",
+			exist:       false,
+			err:         fmt.Errorf("empty project name"),
+		},
```

Comment:
I would drop this case once we add the `contract`. We can also drop the `err` field from the table then.

```suggestion
	}{
```

Reply:
Updated https://github.com/pulumi/pulumi/pull/12798/commits/48756f68b1ec77b93f1ada9bc9dba98a5d1ad806